samza-core/src/main/java/org/apache/samza/table/TableManager.java

```diff
@@ -63,13 +64,17 @@
 
   private boolean localTablesInitialized;
 
+  private final Config config;
```

Comment:
I think not only remote tables have need to access config/taskContext, probably most table implementations do, at least for metrics reason. So I think it makes more sense to make all tables Initable with default blank impl. 

If TM doesn't hold this reference, it will not be able to initialize the table instance in getTable which happens later down the chain.

Reply:
See my earlier comment, the table manager needs to support both per container and per task tables. So can the table provider holds these references?